### PR TITLE
Merged amplitudes

### PR DESCRIPTION
**Pull request by stefan-wallner**

### CODE REVIEW

File: utilities/thisROOTPWA.sh [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+	realpath(){
+		echo "$1"
+	}
+fi
+
+
+ROOTPWA_BUILD="$(dirname "$(dirname "$(realpath "${BASH_SOURCE[0]}")")")"
+export ROOTPWA_BUILD
+
+# usually, but not neccessarily
```
[the remark this answers is not in the document]
Comment (stefan-wallner) [reply utilities/thisROOTPWA.sh]: This means, that the ROOTPWA folder is usually the base older of the build folder, but not necessarily.
I grep it now from the makefile, which should be more robust.

File: pyInterface/module/_config.py [hunk 2]
```diff
@@ -71,9 +70,7 @@ def initialize(self, configFileName):
 			self.intDirectory    = self.getPathFromConfig("general", "intFileDirectory" , configDir + "/ints")
 
 			if self.config.has_option('general', 'limitFilesPerDir'):
-				self.limitFilesPerDir                   = int(self.config.get('general', 'limitFilesPerDir'))
-			else:
-				self.limitFilesPerDir                   = -1
+				pyRootPwa.utils.printWarn("'limitFilesPerDir' options is no longer supported! Ignoring it.")
```
[the remark this answers is not in the document]
Comment (stefan-wallner) [reply pyInterface/module/_config.py]: As just discussed.

#L147: shouldn't this also give a warning message like in #L141?: The messages are printed in the `_binnigInputHandlingForCaseTwoAndThree` function

File: pyInterface/module/_fileManager.py [hunk 3]
```diff
@@ -133,14 +134,13 @@ def initialize(self, configObject):
...
-			self.limitFilesInDir = -1
-			pyRootPwa.utils.printInfo("limit for files per directory set to infinite")
-		else:
-			self.limitFilesInDir = int(configObject.limitFilesPerDir)
-			pyRootPwa.utils.printInfo("limit for files per directory set to " + str(self.limitFilesInDir))
-		self.amplitudeFiles = self._getAmplitudeFilePaths()
-		pyRootPwa.utils.printInfo("number of amplitude files: " + str(len(self.amplitudeFiles)))
+		amplitudeFiles = self._getAmplitudeFilePaths(self.dataFiles.keys(), self.getWaveNameList())
+		# make sure directories exist (create if necessary) and check if they are empty
+		if not os.path.exists(self.amplitudeDirectory):
```
[the remark this answers is not in the document]
Comment (stefan-wallner) [reply pyInterface/module/_fileManager.py]: The amplitude directory was already created in the config object.

File: pyInterface/module/_fileManager.py [hunk 4]
```diff
@@ -215,30 +254,37 @@ def getIntegralFilePath(self, multiBin, eventsType):
...
-			pyRootPwa.utils.printWarn("events type '" + str(eventsType) + "' not in data files.")
+		if eventsType not in self.eventFiles:
+			pyRootPwa.utils.printWarn("events type '" + str(eventsType) + "' not in event files.")
 			return []
 		eventFileIds = []
-		for eventFileId, inputFile in enumerate(self.dataFiles[eventsType]):
-			found = True
-			for variableName in inputFile.multibinBoundaries:
+		for eventFileId, eventFile in enumerate(self.eventFiles[eventsType]):
+			overlays = True # event-file bin overlay with the given multibin
```
Comment (bgrube) [inline pyInterface/module/_fileManager.py]: Maybe "overlap" instead of "overlay"?

**Comment by stefan-wallner** [comment]
Thanks a lot for your comments and the very fast review!

**Comment by bgrube** [comment]
Merged. Thank you!

**Pull request merged**